flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherFailoverITCase.java

```diff
@@ -214,7 +214,7 @@ private TestingDispatcher createRecoveredDispatcher(
         for (JobID jobId : haServices.getJobGraphStore().getJobIds()) {
             jobGraphs.add(haServices.getJobGraphStore().recoverJobGraph(jobId));
         }
-        haServices.setRunningJobsRegistry(new StandaloneRunningJobsRegistry());
+        haServices.setJobResultStore(new EmbeddedJobResultStore());
```

Comment:
For tests it would be helpful to have a default `TestingJobResultStore`. I actually added such an implementation already in [eb78bdb](https://github.com/autophagy/flink/pull/2/commits/eb78bdb22edd0a2ca15b555338287ab47dca8834). Feel free to cherry-pick this commit if you want to have it as part of this PR. I guess, this PR is the right one to cover it. I'm not going to comment on each test where the `TestingJobResultStore` could be used, instead of the `EmbeddedJobResultStore`.